### DISCUSSION TITLE
Feature request: Honor SPDX identifier in LICENSE file

### DISCUSSION THREAD

**Discussion by svenssonaxel** [community/community#39182] [Repositories]
### Select Topic Area

General

### Body

GitHub recognizes some of the more common license texts and derives the SPDX identifier of the repository from there.

SPDX however allows for more expressive license identifiers such as the `OR` operator that gives the user more than one license to choose from. It would be great if GitHub could recognize when an SPDX license identifier is stated explicitly in the LICENSE file and report that for the repository's license.

**Comment by Cryolitia** [comment]
It's important for Linux distribution software repository maintainers. Since more and more software repository enforced to use SPDX identifier like `AGPL-3.0-only` or `AGPL-3.0-or-later` instead of unclear and non-SPDX-indentifier `AGPL-3.0`. 

Now, maintainers could only open issue to ask or search in docs or `README` to determine exactly which SPDX identifier of the licnese that a software choosed.

GitHub should just let a repo's owner to choose a SPDX identifier or upload an uncommon LICENSE file, instead of letting user upload anything called `LICENSE` whatever they want and show a unclear and non-SPDX-indentifier `AGPL-3.0`. 

Related:

- Arch Linux: [Use SPDX license identifiers in PKGBUILDs](https://gitlab.archlinux.org/archlinux/rfcs/-/blob/master/rfcs/0016-spdx-license-identifiers.rst)
- Nixpkgs (NixOS software repo): https://github.com/NixOS/nixpkgs/pull/297173
